module/VuFind/src/VuFind/Controller/AbstractBase.php

```diff
@@ -346,16 +350,11 @@ public function forceLogin($msg = null, $extras = [], $forward = true)
             $msg = 'You must be logged in first';
         }
 
-        // We don't want to return to the lightbox
-        $serverUrl = $this->getServerUrl();
-        $serverUrl = str_replace(
-            ['?layout=lightbox', '&layout=lightbox'],
-            ['?', '&'],
-            $serverUrl
-        );
+        // store parent url of lightboxes
+        $extras['lightboxParent'] = urldecode($this->getRequest()->getQuery('lightboxParent'));
```

Comment:
Why does this need to be URL decoded? Where is it getting encoded? Maybe a comment would be in order to help clarify what is going on here.

Reply:
It does not need to be decoded. I removed it.